src/show.jl

```diff
@@ -0,0 +1,174 @@
+function Base.show(io::IO, fid::File)
+    if isvalid(fid)
+        print(io, "HDF5.File: ", fid.filename)
+    else
+        print(io, "HDF5.File: (closed) ", fid.filename)
+    end
+end
+
+function Base.show(io::IO, g::Group)
+    if isvalid(g)
+        print(io, "HDF5.Group: ", name(g), " (file: ", g.file.filename, ")")
+    else
+        print(io, "HDF5.Group: (invalid)")
+    end
+end
+
+function Base.show(io::IO, prop::Properties)
+    if prop.class == H5P_DEFAULT
+        print(io, "HDF5.Properties: default class")
+    elseif isvalid(prop)
+        print(io, "HDF5.Properties: ", h5p_get_class_name(prop.class), " class")
+    else
+        print(io, "HDF5.Properties: (invalid)")
+    end
+end
+
+function Base.show(io::IO, dset::Dataset)
+    if isvalid(dset)
+        print(io, "HDF5.Dataset: ", name(dset), " (file: ", dset.file.filename, " xfer_mode: ", dset.xfer.id, ")")
+    else
+        print(io, "HDF5.Dataset: (invalid)")
+    end
+end
+
+function Base.show(io::IO, attr::Attribute)
+    if isvalid(attr)
+        print(io, "HDF5.Attribute: ", name(attr))
+    else
+        print(io, "HDF5.Attribute: (invalid)")
+    end
+end
+function Base.show(io::IO, attr::Attributes)
+    print(io, "Attributes of ", attr.parent)
+end
+
+function Base.show(io::IO, dtype::Datatype)
+    print(io, "HDF5.Datatype: ")
+    if isvalid(dtype)
+        h5t_committed(dtype) && print(io, name(dtype), " ")
+        print(io, h5lt_dtype_to_text(dtype))
+    else
+        # Note that h5i_is_valid returns `false` on the built-in datatypes (e.g.
+        # H5T_NATIVE_INT), apparently because they have refcounts of 0 yet are always
+        # valid. Just temporarily turn off error printing and try the call to probe if
+        # dtype is valid since H5LTdtype_to_text special-cases all of the built-in types
+        # internally.
+        old_func, old_client_data = h5e_get_auto(H5E_DEFAULT)
+        h5e_set_auto(H5E_DEFAULT, C_NULL, C_NULL)
+        local text
+        try
+            text = h5lt_dtype_to_text(dtype)
+        catch
+            text = "(invalid)"
+        finally
+            h5e_set_auto(H5E_DEFAULT, old_func, old_client_data)
+        end
+        print(io, text)
+    end
+end
+
+"""
+    SHOW_TREE = Ref{Bool}(true)
+
+Configurable option to control whether the default `show` for HDF5 objects is printed
+using `show_tree` or not.
+"""
+const SHOW_TREE = Ref{Bool}(true)
+"""
+    SHOW_TREE_ICONS = Ref{Bool}(true)
+
+Configurable option to control whether emoji icons (`true`) or a plain-text annotation
+(`false`) is used to indicate the object type by `show_tree`.
+"""
+const SHOW_TREE_ICONS = Ref{Bool}(true)
+
+function Base.show(io::IO, ::MIME"text/plain", obj::Union{File,Group,Dataset,Attributes,Attribute})
+    if SHOW_TREE[]
+        show_tree(io, obj)
+    else
+        show(io, obj)
+    end
+end
+
+function _tree_icon(obj)
+    if SHOW_TREE_ICONS[]
+        return obj isa Attribute ? "🏷️ " :
```

Comment:
Minor nit, might be better to remove the space after the emoji here and add in the `print` line  in the `_tree....` function.